flight/Modules/Actuator/actuator.c

```diff
@@ -65,43 +64,57 @@
 
 #define TASK_PRIORITY PIOS_THREAD_PRIO_HIGHEST
 #define FAILSAFE_TIMEOUT_MS 100
+
+#ifndef MAX_MIX_ACTUATORS
 #define MAX_MIX_ACTUATORS ACTUATORCOMMAND_CHANNEL_NUMELEM
-#define MULTIROTOR_MIXER_UPPER_BOUND 128
+#endif
+
+DONT_BUILD_IF(ACTUATORSETTINGS_TIMERUPDATEFREQ_NUMELEM > PIOS_SERVO_MAX_BANKS, TooManyServoBanks);
+DONT_BUILD_IF(MAX_MIX_ACTUATORS > ACTUATORCOMMAND_CHANNEL_NUMELEM, TooManyMixers);
+DONT_BUILD_IF((MIXERSETTINGS_MIXER1VECTOR_NUMELEM - MIXERSETTINGS_MIXER1VECTOR_ACCESSORY0) < MANUALCONTROLCOMMAND_ACCESSORY_NUMELEM, AccessoryMismatch);
+
+#define MIXER_SCALE 128
 
 // Private types
 
 // Private variables
 static struct pios_queue *queue;
 static struct pios_thread *taskHandle;
 
-static bool flightStatusUpdated = true;
-static bool manualControlCommandUpdated = true;
-
 // used to inform the actuator thread that actuator / mixer settings are updated
 // set true to ensure they're fetched on first run
+static volatile bool flight_status_updated = true;
+static volatile bool manual_control_cmd_updated = true;
 static volatile bool actuator_settings_updated = true;
 static volatile bool mixer_settings_updated = true;
 
-// The actual mixer settings data, pulled at the top of the actuator thread
-static MixerSettingsData mixerSettings;
+static MixerSettingsMixer1TypeOptions types_mixer[MAX_MIX_ACTUATORS];
+
+/* In the mixer, a row consists of values for one output actuator.
+ * A column consists of values for scaling one axis's desired command.
+ */
+
+static float motor_mixer[MAX_MIX_ACTUATORS * MIXERSETTINGS_MIXER1VECTOR_NUMELEM];
 
-// Ditto, for the actuator settings.
+/* These are various settings objects used throughout the actuator code */
 static ActuatorSettingsData actuatorSettings;
+static SystemSettingsAirframeTypeOptions airframe_type;
+
+static float curve1[MIXERSETTINGS_THROTTLECURVE1_NUMELEM];
```

Comment:
To save memory, we no longer keep around the un-transformed mixer settings at global scope.  There are a few things we need to pull out explicitly.

Note that the current use of curve1 source for throttle curve "feel" is problematic but unlikely to be fixed for neat.